pkg/controller/node/nodecontroller.go

```diff
@@ -368,6 +371,28 @@ func (nc *NodeController) maybeDeleteTerminatingPod(obj interface{}) {
 	}
 }
 
+// cleanupOrphanedPods deletes pods that are bound to nodes that don't
+// exist.
+func (nc *NodeController) cleanupOrphanedPods() {
+	pods, err := nc.podStore.List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+
+	for _, pod := range pods {
+		if pod.Spec.NodeName == "" {
+			continue
+		}
+		if _, exists, _ := nc.nodeStore.Store.GetByKey(pod.Spec.NodeName); exists {
+			continue
+		}
+		if err := nc.forcefullyDeletePod(pod); err != nil {
```

Comment:
My intuition is that this race is pretty minor and less significant than the bug it fixes, but our intuition is often wrong about this stuff, and on an antagonistically loaded machine, you could imagine being in some kind of steady state forever where the controllers fight each other.

But there's a non-controller behavior that's also new.  If I manually specify a host in my single pod (no controller), but make a typo, my pod disappears and never runs.  That'll seem weird.  It's also possible for the user to submit the pod in the same race window (hard to imagine this really happening) and the same thing happens.

Suggestions for fixing the race... I'll think about it.  Concurrency + global state + caching lends itself to this.